hexrd/ui/hand_drawn_mask_dialog.py

```diff
@@ -42,12 +43,21 @@ def __init__(self, canvas, parent):
     def setup_connections(self):
         self.ui.accepted.connect(self.accepted)
         self.ui.rejected.connect(self.rejected)
-        self.bp_id = self.canvas.mpl_connect('button_press_event',
-                                             self.button_pressed)
-        self.enter_id = self.canvas.mpl_connect('axes_enter_event',
-                                                self.axes_entered)
-        self.exit_id = self.canvas.mpl_connect('axes_leave_event',
-                                               self.axes_exited)
+        for canvas in self.parent().image_tab_widget.active_canvases:
+            press = canvas.mpl_connect('button_press_event',
+                                                self.button_pressed)
+            enter = canvas.mpl_connect('axes_enter_event',
+                                                    self.axes_entered)
+            exit = canvas.mpl_connect('axes_leave_event',
+                                                self.axes_exited)
+            self.canvas_ids.append([press, enter, exit])
+        self.parent().image_tab_widget.tabBarClicked.connect(
```

Comment:
Same here. I think it's better for this connection to be made in the parent object instead. That can be done in the `on_action_edit_apply_hand_drawn_mask_triggered` function.

Reply:
You can also change the `canvas` there too on this object when the tabbed view changes.

Reply:
Connection moved.

Reply:
Sounds good! But I think you need to push your changes? I don't see them here.

Reply:
Pushed in the latest WIP (questions in comment)